src/main/java/datawave/microservice/query/cachedresults/CachedResultsQueryService.java

```diff
@@ -0,0 +1,1364 @@
+    private final QueryLogicFactory queryLogicFactory;
+    private final QueryStorageCache queryStorageCache;
+    private final ResponseObjectFactory responseObjectFactory;
+    private final MarkingFunctions markingFunctions;
+    // Note: CachedResultsQueryParameters needs to be request scoped
+    private final CachedResultsQueryParameters cachedResultsQueryParameters;
+    private final String fieldDefinitions;
+    private final String preparedFields;
+    private final String preparedValues;
+    
+    public CachedResultsQueryService(CachedResultsQueryProperties cachedResultsQueryProperties, JdbcTemplate cachedResultsJdbcTemplate,
+                    CachedResultsQueryCache cachedResultsQueryCache, QueryService queryService, AuditClient auditClient, SecurityMarking securityMarking,
+                    QueryLogicFactory queryLogicFactory, QueryStorageCache queryStorageCache, ResponseObjectFactory responseObjectFactory,
+                    MarkingFunctions markingFunctions, CachedResultsQueryParameters cachedResultsQueryParameters) {
+        this.cachedResultsQueryProperties = cachedResultsQueryProperties;
+        this.cachedResultsJdbcTemplate = cachedResultsJdbcTemplate;
+        this.cachedResultsQueryCache = cachedResultsQueryCache;
+        this.queryService = queryService;
+        this.auditClient = auditClient;
+        this.securityMarking = securityMarking;
+        this.queryLogicFactory = queryLogicFactory;
+        this.queryStorageCache = queryStorageCache;
+        this.responseObjectFactory = responseObjectFactory;
+        this.markingFunctions = markingFunctions;
+        this.cachedResultsQueryParameters = cachedResultsQueryParameters;
+        this.fieldDefinitions = IntStream.range(0, cachedResultsQueryProperties.getNumFields()).mapToObj(x -> FIELD + x + " LONGTEXT")
+                        .collect(Collectors.joining(", "));
+        this.preparedFields = IntStream.range(0, cachedResultsQueryProperties.getNumFields()).mapToObj(x -> FIELD + x).collect(Collectors.joining(", "));
+        this.preparedValues = Stream.generate(() -> "?").limit(cachedResultsQueryProperties.getNumFields()).collect(Collectors.joining(", "));
+        initializeTableTemplate();
+    }
+    
+    private void initializeTableTemplate() {
+        // @formatter:off
+        String createTableTemplate = cachedResultsQueryProperties.getStatementTemplates().getCreateTableTemplate()
+                .replace(FIELD_DEFINITIONS_PLACEHOLDER, fieldDefinitions);
+        // @formatter:on
+        try {
+            cachedResultsJdbcTemplate.execute(createTableTemplate);
+        } catch (DataAccessException e) {
+            log.error("Unable to create 'create table' template with statement: {}", createTableTemplate, e);
+            throw e;
+        }
+    }
+    
+    // Prior to the LOAD call, the user defines a query via the query API
+    
+    // TODO: JWO: Add a method to manually remove a query's results from the cache
```

Comment:
Maybe it's worth addressing the TODOs or documenting them in issues.

Reply:
yup, im going to go through them today.  my goal is to remove any that are in the main datawave repo, but some may remain in the query microservice repos since those are still sort of in development.

Reply:
went through all of the TODOs and resolved them one way or another.